src/app/stacktrace_win.h

```diff
@@ -108,6 +112,68 @@ BOOL CALLBACK straceWin::EnumModulesCB(LPCSTR ModuleName, DWORD64 BaseOfDll, PVO
 }
 
 
+/**
+* Cuts off leading 'dir' path from 'file' path, otherwise leaves it unchanged
+* returns true iff 'dir' is an ancestor of 'file'
+*/
+bool straceWin::makeRelativePath(const QString& dir, QString& file)
+{
+    auto d = QDir::toNativeSeparators(QDir(dir).canonicalPath());
+    auto f = QDir::toNativeSeparators(QFileInfo(file).canonicalFilePath());
+
+    // append separator at the end of dir
+    auto separator = QDir::separator();
+    if (!d.isEmpty() && (d[d.length() - 1] != separator))
+        d += separator;
+
+    if (f.startsWith(d, Qt::CaseInsensitive)) {
+        f.remove(0, d.length());
+        file.swap(f);
+
+        return true;
+    }
+
+    return false;
+}
+
+QString straceWin::getSourcePathAndLineNumber(HANDLE hProcess, DWORD64 addr)
+{
+    IMAGEHLP_LINE64 line{};
+    line.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
+    DWORD dwDisplacement{};
+
+    if (SymGetLineFromAddr64(hProcess, addr, &dwDisplacement, &line)){
+        QString path(line.FileName);
+
+
+#if defined STACKTRACE_WIN_PROJECT_PATH || defined STACKTRACE_WIN_MAKEFILE_PATH
+
+#define STACKTRACE_WIN_QUOTE(x)  #x
+#define STACKTRACE_WIN_STRING(x)  STACKTRACE_WIN_QUOTE(x)
+
+        //prune leading project directory path or build target directory path
+
+        bool success = false;
+#ifdef STACKTRACE_WIN_PROJECT_PATH
+        QString projectPath(STACKTRACE_WIN_STRING(STACKTRACE_WIN_PROJECT_PATH));
+        success = makeRelativePath(projectPath, path);
+#endif
+
+#ifdef STACKTRACE_WIN_MAKEFILE_PATH
+        if (!success) {
+            QString targetPath(STACKTRACE_WIN_STRING(STACKTRACE_WIN_MAKEFILE_PATH));
+            makeRelativePath(targetPath, path);
+        }
+#endif
+#endif
+        return QString("%1 : %2")
+            .arg(path)
+            .arg(line.LineNumber);
+    }
+
+    return{};
```

Comment:
It is generally a perversion. Everything has its place. I would understand it if you return a structure, array or something like that...